src/ivc/incrementally_verifiable_computation.rs

```diff
@@ -112,4 +240,49 @@ where
         // TODO #31
         todo!("Logic at `RecursiveSNARK::verify`")
     }
+
+    fn prepare_primary_td<const T: usize, RP1>(
+        k_table_size: u32,
+        instance: [C1::Scalar; 2],
+    ) -> (TableData<C1::Scalar>, StepConfig<A1, C1::Scalar, SC1, T>)
+    where
+        RP1: ROPair<C1::Scalar, Config = MainGateConfig<T>>,
+    {
+        let mut primary_td = TableData::new(k_table_size, instance.to_vec());
+        let config = primary_td.prepare_assembly(
+            <crate::ivc::step_folding_circuit::StepFoldingCircuit<
+                '_,
+                A1,
+                C2,
+                SC1,
+                RP1::OnCircuit,
+                T,
+            > as StepCircuit<A1, C1::Scalar>>::configure,
+        );
+
+        (primary_td, config)
+    }
+
+    fn prepare_secondary_td<const T: usize, RP2>(
```

Comment:
since prepare_primary_td and prepare_secondary_td are similar, can we use generic to make them one if it doesn't make generics too complicated?

Reply:
Yeah, the generic is getting unnecessarily complicated